frappe/integrations/doctype/paytm_settings/checksum.py

```diff
@@ -0,0 +1,142 @@
+import base64
+import string
+import random
+import hashlib
+
+from Crypto.Cipher import AES
+
+
+IV = "@@@@&&&&####$$$$"
+BLOCK_SIZE = 16
+
+
+def generate_checksum(param_dict, merchant_key, salt=None):
+	params_string = __get_param_string__(param_dict)
+	salt = salt if salt else __id_generator__(4)
+	final_string = '%s|%s' % (params_string, salt)
+
+	hasher = hashlib.sha256(final_string.encode())
+	hash_string = hasher.hexdigest()
+
+	hash_string += salt
+
+	return __encode__(hash_string, IV, merchant_key)
+
+
+def generate_refund_checksum(param_dict, merchant_key, salt=None):
+	for i in param_dict:
+		if("|" in param_dict[i]):
+			param_dict = {}
+			exit()
+	params_string = __get_param_string__(param_dict)
+	salt = salt if salt else __id_generator__(4)
+	final_string = '%s|%s' % (params_string, salt)
+
+	hasher = hashlib.sha256(final_string.encode())
+	hash_string = hasher.hexdigest()
+
+	hash_string += salt
+
+	return __encode__(hash_string, IV, merchant_key)
+
+
+def generate_checksum_by_str(param_str, merchant_key, salt=None):
+	params_string = param_str
+	salt = salt if salt else __id_generator__(4)
+	final_string = '%s|%s' % (params_string, salt)
+
+	hasher = hashlib.sha256(final_string.encode())
+	hash_string = hasher.hexdigest()
+
+	hash_string += salt
+
+	return __encode__(hash_string, IV, merchant_key)
+
+
+def verify_checksum(param_dict, merchant_key, checksum):
+	# Remove checksum
+	if 'CHECKSUMHASH' in param_dict:
+		param_dict.pop('CHECKSUMHASH')
+
+	# Get salt
+	paytm_hash = __decode__(checksum, IV, merchant_key)
+	salt = paytm_hash[-4:]
+	calculated_checksum = generate_checksum(
+		param_dict, merchant_key, salt=salt)
+	return calculated_checksum == checksum
+
+
+def verify_checksum_by_str(param_str, merchant_key, checksum):
```

Comment:
you probably do not need to write the same method twice; different for str and dict